mergify_engine/tests/functional/base.py

```diff
@@ -622,6 +625,18 @@ async def create_pr(
             await self.git("commit", "--no-edit", "-m", f"{title}, moved")
         await self.git("push", "--quiet", base_repo, branch)
 
+        if verified:
+            temporary_folder = tempfile.mkdtemp()
+            os.environ["GNUPGHOME"] = temporary_folder
```

Comment:
You should not change the environment for the whole pytest process, but just for the ongoing test or just for the commands that require them. (see `env=` of subprocess module)

Reply:
You must not modify `os.environ`.